cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/AbstractSchedulerService.java

```diff
@@ -185,8 +190,16 @@ public void schedule(Id.Program programId, SchedulableProgramType programType, I
         throw Throwables.propagate(e);
       }
       for (Schedule schedule : schedules) {
-        String scheduleName = schedule.getName();
-        String cronEntry = schedule.getCronEntry();
+        Preconditions.checkArgument(schedule instanceof TimeSchedule || !(schedule instanceof StreamSizeSchedule));
```

Comment:
this is a bit awkward. And it breaks as soon as we add another type of schedules. But I think it goes away if the configurer replaces all plain Schedules with TimeSchedules.